main.rb

```diff
@@ -0,0 +1,307 @@
+    end
+
+    def my_any_array_value_or_no_parameter?(mee, arg = [])
+      if arg == []
+        i = 0
+        mee.my_each { |x| i += 1 if [false, nil].include? x }
+        return (i != mee.size) # no arg (all are false) = flase
+      else
+        mee.my_each { |x| return true if x == arg }
+      end
+      false
+    end
+
+    def my_any_array_parameter?(mee, arg = [])
+      ret = false
+      if arg.class == Class
+        mee.my_each { |x| ret = true if x.class == arg }
+      elsif arg.class == Regexp
+        mee.my_each { |x| ret = true if arg.match? x.to_s }
+      else
+        ret = my_any_array_value_or_no_parameter?(mee, arg)
+      end
+      ret
+    end
+
+    def my_any?(arg = [])
+      me = self
+      ret = false
+      if block_given?
+        ret = my_any_block?(me) { |x, y| yield(x, y) }
+      else
+        return false if me.is_a?(Hash)
+
+        ret = my_any_array_parameter?(me, arg)
+      end
+      ret
+    end
+
+    def my_none_block?(mee)
+      if mee.is_a?(Array)
+        mee.my_each { |x| return false if yield(x) }
+      else # if mee.is_a?(Hash)
+        mee.my_each { |x, y| return false if yield(x, y) }
+      end
+      true
+    end
+
+    def my_none_array_value_or_no_parameter?(mee, arg = [])
+      if arg == []
+        mee.my_each { |x| return false if x } # false until all val are false
+      else
+        mee.my_each { |x| return false if x == arg }
+      end
+      true
+    end
+
+    def my_none_array_parameter?(mee, arg = [])
+      ret = true
+      if arg.class == Class
+        mee.my_each { |x| ret = false if x.class == arg }
+      elsif arg.class == Regexp
+        mee.my_each { |x| ret = false if arg.match? x.to_s }
+      else
+        ret = my_none_array_value_or_no_parameter?(mee, arg)
+      end
+      ret
+    end
+
+    def my_none?(arg = [])
+      me = self
+      ret = true
+      if block_given?
+        ret = my_none_block?(me) { |x, y| yield(x, y) }
+      else
+        return false if me.is_a?(Hash)
+
+        ret = my_none_array_parameter?(me, arg)
+      end
+      ret
+    end
+  end
+
+  # contains "my_count, my_map" methods
+  module EnumerableInside3
+    # my_count_block_array
+    def a_c1(mee, counter)
+      mee.my_each { |x| counter += 1 if yield(x) }
+      counter
+    end
+
+    # my_count_block_hash
+    def h_c1(mee, counter)
+      mee.my_each { |x, y| counter += 1 if yield(x, y) }
+      counter
+    end
+
+    # my_count_parameter_array
+    def a_c2(mee, counter, par)
+      mee.my_each { |x| counter += 1 if x == par }
+      counter
+    end
+
+    # my_count_parameter_hash
+    def h_c2(mee, counter, par)
+      mee.my_each { |_x, y| counter += 1 if y == par }
+      counter
+    end
+
+    def my_count(val = nil)
+      me = self
+      c = 0
+      return me.size if val.nil? && block_given? == false
+
+      c = if me.is_a?(Array)
+            block_given? ? a_c1(me, c) { |x| yield(x) } : a_c2(me, c, val)
+          else # if me.is_a?(Hash)
```

Comment:
* [ ] Make sure you don't have such ```# if me.is_a?(Hash)``` comments in your code. I understand perfectly what the comment sought to achieve but it more of a code form. Should you want to have a comment in your code, let it be plain English.